api/app/applicationModel/api.py

```diff
@@ -79,18 +88,67 @@ def get(self):
             for s in form.sections:
                 s.questions = []
                 for q in questions:
-                    if(q.section_id == s.id):
+                    if (q.section_id == s.id):
                         s.questions.append(q)
 
-            if (form): 
+            if (form):
                 return marshal(form, self.form_fields)
             else:
-                LOGGER.warn("Event not found for event_id: {}".format(args['event_id'])) 
+                LOGGER.warn("Event not found for event_id: {}".format(args['event_id']))
                 return EVENT_NOT_FOUND
 
         except SQLAlchemyError as e:
             LOGGER.error("Database error encountered: {}".format(e))
             return DB_NOT_AVAILABLE
-        except: 
+        except:
             LOGGER.error("Encountered unknown error: {}".format(traceback.format_exc()))
             return DB_NOT_AVAILABLE
+
+    @auth_required
+    def post(self):
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+
+        event = db.session.query(Event).get(event_id)
+        if not event:
+            return EVENT_NOT_FOUND
+
+        user_id = g.current_user["id"]
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_admin:
+            return FORBIDDEN
+
+        app_form = app_repository.get_by_event_id(event_id)
+        if not app_form:
+            app_form = ApplicationForm, Section, Question
+            app_form = app_repository.add(app_form)
+
+        db.session.commit()
+
+        app_id = app_repository.get_by_id(app_form.id)
+        return app_form(app_id), 201
+
+
+    @auth_required
+    def put(self):
+        args = self.req_parser.parse_args()
+        event_id = args['event_id']
+
+        event = db.session.query(Event).get(event_id)
+        if not event:
+            return EVENT_NOT_FOUND
+
+        user_id = g.current_user["id"]
+        current_user = user_repository.get_by_id(user_id)
+        if not current_user.is_admin:
+            return FORBIDDEN
+
+        app_form = app_repository.get_by_event_id(event_id)
+        try:
+            app_form.update(ApplicationForm, Section, Question)
+
+        except IntegrityError:
+            LOGGER.error("Application with id: {} does not exist".format(app_form.id))
```

Comment:
Actually an integrity error would occur if you're trying to insert an ID that *already* exists?